File: solutions/2026-07-30/java/2026-07-30-minimum-number-of-pushes-to-type-word-i.java
```java
/**
 * Problem: Minimum Number of Pushes to Type Word I
 * Link: https://leetcode.com/problems/minimum-number-of-pushes-to-type-word-i/
 *
 * Approach:
 * To minimize the total number of pushes, we should assign the most frequent characters
 * (or in this case, all characters since they are distinct and we want to minimize total pushes for a given word)
 * to keys that require fewer pushes. The keys that require fewer pushes are those that can be mapped
 * to a smaller number of characters on a standard telephone keypad.
 *
 * On a telephone keypad, keys 2-9 can be mapped. Each key can hold up to 8 characters (a-z have 26 letters).
 * The keys that require 1 push are those mapped to the first character on their respective keys.
 * The keys that require 2 pushes are those mapped to the second character.
 * And so on.
 *
 * Since all letters in the input `word` are distinct and we want to minimize the total pushes for *that specific word*,
 * we should assign the first 8 characters of the word to keys that require 1 push.
 * The next 8 characters to keys that require 2 pushes.
 * The remaining characters (up to 8) to keys that require 3 pushes.
 *
 * So, if the word length is `n`:
 * - The first 8 characters will contribute 1 push each to the total.
 * - The next 8 characters will contribute 2 pushes each.
 * - The remaining `n - 16` characters (if `n > 16`) will contribute 3 pushes each.
 *
 * This can be generalized:
 * For characters at index `i` (0-indexed) in the sorted word (which is the input word itself as it's distinct):
 * - If `i < 8`, cost is 1.
 * - If `8 <= i < 16`, cost is 2.
 * - If `16 <= i < 24`, cost is 3.
 *
 * The total number of pushes is the sum of pushes for each character.
 *
 * Time Complexity: O(N), where N is the length of the word. We iterate through the word once to calculate the pushes.
 *                  The problem constraints state N <= 26, so this is effectively O(1).
 * Space Complexity: O(1), as we only use a few variables to store the total pushes.
 */
class Solution {
    public int minimumPushes(String word) {
        // The length of the word. Since all letters are distinct and we want to minimize
        // pushes for this specific word, we can directly use the word's length.
        int n = word.length();
        // Initialize the total number of pushes to 0.
        int totalPushes = 0;

        // We distribute the characters of the word across keys requiring 1, 2, and 3 pushes.
        // Keys 2-5 require 1 push (4 keys * 1 char/key = 4 chars) -> This is incorrect based on typical keypads.
        // Standard keypad mapping:
        // Key 2: a, b, c (3 chars)
        // Key 3: d, e, f (3 chars)
        // Key 4: g, h, i (3 chars)
        // Key 5: j, k, l (3 chars)
        // Key 6: m, n, o (3 chars)
        // Key 7: p, q, r, s (4 chars)
        // Key 8: t, u, v (3 chars)
        // Key 9: w, x, y, z (4 chars)
        // Total available slots: 3+3+3+3+3+4+3+4 = 26.
        //
        // The problem statement implies we can remap keys to distinct collections.
        // The example shows that we assign characters greedily to minimize pushes.
        // For the first 8 characters, assign them to 8 different keys, each requiring 1 push.
        // For the next 8 characters, assign them to 8 different keys, each requiring 2 pushes.
        // For the remaining characters, assign them to 8 different keys, each requiring 3 pushes.
        // This implies there are enough keys (2-9, which is 8 keys) to support this distribution strategy.

        // Calculate pushes for the first 8 characters (if they exist). Each requires 1 push.
        // The number of characters in this group is min(n, 8).
        int charsFor1Push = Math.min(n, 8);
        totalPushes += charsFor1Push * 1;

        // If there are more than 8 characters, calculate pushes for the next 8 characters.
        // These characters are from index 8 up to index 15 (exclusive).
        if (n > 8) {
            // The number of characters in this group is min(n - 8, 8).
            int charsFor2Pushes = Math.min(n - 8, 8);
            totalPushes += charsFor2Pushes * 2;
        }

        // If there are more than 16 characters, calculate pushes for the remaining characters.
        // These characters are from index 16 up to the end of the word.
        if (n > 16) {
            // The number of characters in this group is n - 16.
            int charsFor3Pushes = n - 16;
            totalPushes += charsFor3Pushes * 3;
        }

        // Return the calculated total minimum number of pushes.
        return totalPushes;
    }
}
```

File: solutions/2026-07-30/cpp/2026-07-30-minimum-number-of-pushes-to-type-word-i.cpp
```cpp
// Problem: Minimum Number of Pushes to Type Word I
// Link: https://leetcode.com/problems/minimum-number-of-pushes-to-type-word-i/
//
// Approach:
// The problem asks us to find the minimum number of pushes to type a given word
// where each character is distinct. We can remap keys 2-9 (which can hold up to 8 characters)
// to minimize the total pushes. To minimize the total pushes, we should assign the
// most frequent characters to keys that require fewer pushes. Since all characters
// in the input `word` are distinct, each character appears exactly once.
//
// The most efficient way to assign characters is to put as many characters as possible
// on keys that require only 1 push. There are 8 keys (2-9) that can be remapped.
// We can put up to 8 characters on these keys with 1 push each.
// The next most efficient assignment would be to put characters on keys that require
// 2 pushes. We can do this for up to 8 characters (using the 8 keys again).
// And so on.
//
// Given that the input `word` has distinct characters, we can think of it as
// assigning each character to a specific key. To minimize pushes, we should assign
// characters greedily. The first 8 characters of the word should be assigned to keys
// that require 1 push. The next 8 characters should be assigned to keys that require
// 2 pushes. The next 8 characters (if any) should be assigned to keys that require
// 3 pushes.
//
// For example, if the word has length L:
// - The first min(L, 8) characters will require 1 push each.
// - The next min(L - 8, 8) characters will require 2 pushes each.
// - The next min(L - 16, 8) characters will require 3 pushes each.
//
// We can iterate through the length of the word and calculate the contribution of
// each character to the total pushes.
//
// Time Complexity: O(N), where N is the length of the `word`. We iterate through the word once.
// Space Complexity: O(1), as we only use a few variables to store the total pushes and loop counters.
class Solution {
public:
    int minimumPushes(std::string word) {
        int n = word.length();
        int totalPushes = 0;
        // We can assign up to 8 characters to keys that require 1 push.
        // The next 8 characters to keys that require 2 pushes, and so on.
        //
        // For characters from index 0 to 7 (first 8), each requires 1 push.
        // For characters from index 8 to 15 (next 8), each requires 2 pushes.
        // For characters from index 16 to 23 (next 8), each requires 3 pushes.
        //
        // This logic can be implemented by observing the number of characters
        // that fall into each "push level".
        //
        // Level 1: First 8 characters. Contribution: 8 * 1
        // Level 2: Next 8 characters. Contribution: 8 * 2
        // Level 3: Next 8 characters. Contribution: 8 * 3
        // ... and so on.
        //
        // We can use a loop to calculate this.
        // The number of characters that fall into a certain push level `p`
        // is limited by 8 (the number of keys available for remapping)
        // and the remaining characters in the word.
        //
        // Example: word.length() = 10
        // Chars 0-7: 8 characters. Pushes = 8 * 1 = 8. Remaining chars = 10 - 8 = 2.
        // Chars 8-9: 2 characters. Pushes = 2 * 2 = 4. Remaining chars = 2 - 2 = 0.
        // Total pushes = 8 + 4 = 12.
        //
        // This can be calculated as:
        // Iterate through push levels p = 1, 2, 3, ...
        // For each level p, the number of characters assigned to this level is:
        // `numCharsAtThisLevel = min(remaining_chars, 8)`
        // `totalPushes += numCharsAtThisLevel * p`
        // `remaining_chars -= numCharsAtThisLevel`
        // Stop when `remaining_chars` is 0.

        int remainingChars = n;
        for (int pushes = 1; remainingChars > 0; ++pushes) {
            // The number of characters we can assign to the current push level.
            // It's either 8 (max characters per key) or the number of remaining characters,
            // whichever is smaller.
            int charsForThisPushLevel = std::min(remainingChars, 8);

            // Add the total pushes for this level.
            totalPushes += charsForThisPushLevel * pushes;

            // Decrease the count of remaining characters.
            remainingChars -= charsForThisPushLevel;
        }

        return totalPushes;
    }
};
```